src/Collapse/utilities/distance_matrix_edge_collapse_rips_persistence.cpp

```diff
@@ -0,0 +1,177 @@
+         "Maximum allowed dimension of the Rips complex.")
+        ("input_file_name,i", po::value<std::string>(&csv_matrix_file),
+         "The input file.")
+        ("filediag,o", po::value<std::string>(&filediag),
+         "The output file.");
+
+  po::options_description all;
+  all.add(visible);
+  po::variables_map vm;
+  po::store(po::command_line_parser(argc, argv).options(all).run(), vm);
+  po::notify(vm);
+  if (vm.count("help")) {
+    std::cout << std::endl;
+    std::cout << "Computes rips complexes of different threshold values, to 'end_thresold' from a n random uniform "
+                 "point_vector on a selected manifold, . \n";
+    std::cout << "Strongly collapses all the rips complexes and output the results in out_file. \n";
+    std::cout << "The experiments are repeted 'repete' num of times for each threshold value. \n";
+    std::cout << "type -m for manifold options, 's' for uni sphere, 'b' for unit ball, 'f' for file. \n";
+    std::cout << "type -i 'filename' for Input file option for exported point sample. \n";
+    std::cout << std::endl << std::endl;
+    std::cout << "Usage: " << argv[0] << " [options]" << std::endl << std::endl;
+    std::cout << visible << std::endl;
+    std::abort();
+  }
+}
+
+void program_options(int argc, char* argv[], std::string& csv_matrix_file, std::string& filediag,
+                     Filtration_value& threshold, int& dim_max, int& p, Filtration_value& min_persistence);
+
+int main(int argc, char* argv[]) {
+  std::string csv_matrix_file;
+  std::string filediag;
+  Filtration_value threshold;
+  int dim_max = 2;
+  int p;
+  Filtration_value min_persistence;
+
+  program_options(argc, argv, csv_matrix_file, filediag, threshold, dim_max, p, min_persistence);
+
+  Distance_matrix distances;
+
+  distances = Gudhi::read_lower_triangular_matrix_from_csv_file<Filtration_value>(csv_matrix_file);
+  std::cout << "Read the distance matrix succesfully, of size: " << distances.size() << std::endl;
+
+  Proximity_graph proximity_graph = Gudhi::compute_proximity_graph<Simplex_tree>(boost::irange((size_t)0,
+                                                                                               distances.size()),
+                                                                                 threshold,
+                                                                                 [&distances](size_t i, size_t j) {
+                                                                                   return distances[j][i];
+                                                                                 });
+
+  // Now we will perform filtered edge collapse to sparsify the edge list edge_t.
+  Flag_complex_sparse_matrix flag_complex(proximity_graph);
+
+  Simplex_tree stree;
+  flag_complex.filtered_edge_collapse(
+    [&stree](std::vector<Vertex_handle> edge, Filtration_value filtration) {
+        // insert the 2 vertices with a 0. filtration value just like a Rips
```

Comment:
IIUC edge collapses preserve the 0-skeleton. It seems that it would be better to insert first all the vertices before doing the collapses, and then the callback can insert just the edge.

Reply:
If I would have read this comment more carefully it would have fixed my bug :disappointed: